cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -1,33 +1,48 @@
-from random import sample
-
 import pytest
+from cfme.middleware import get_random_list
 from cfme.middleware.deployment import MiddlewareDeployment
+from cfme.middleware.server import MiddlewareServer
 from utils import testgen
 from utils.version import current_version
 
 pytestmark = [
-    pytest.mark.usefixtures('single_middleware_provider'),
+    pytest.mark.usefixtures('setup_provider'),
```

Comment:
@mfalesni can we have several middleware providers configured in yaml file, and have them in tests?
for instance here: "def test_hawkular_crud(provider):" to have set of providers?